### DISCUSSION TITLE
hx-swap="empty" feature

### DISCUSSION THREAD

**Discussion by rag57** [bigskysoftware/htmx#1379] [Ideas]
I need a cancel function.
I try with hx-swap="delete".
But this will delete my #target in the template and hx-target no longer work.

With an empty-feature I could easy-going realize a cancel configuration.

Any thoughts?

**Comment by jreviews** [comment]
Have you tried `hx-swap="none"`?

https://htmx.org/attributes/hx-swap/

**Comment by andryyy** [comment]
You can also return a 204 status code, which will prevent swaps.

**Reply by rag57** [reply]
But I need to swap to reset the hx-target="#berichtsform"

**Reply by andryyy** [reply]
You may need to provide an example for what exactly you need. Or what a "cancel function" is.

Could be a reset() in JS to reset a form. Or a simple reswap of the initial form to show it in its raw state. Perhaps _hyperscript could be used to reset values. You could use init to save the initial values and "on event" restore them.

Is it like a reset button for each element in a form? _hs is probably the way to go then.

**Reply by andryyy** [reply]
You can also send Reswap and Retarget headers to change swap methods and targets depending on the result of the processing.

**Reply by rag57** [reply]
1. Step

<-- bericht.html -->
....

`{% block content %}`

`<div>`
 `       <button type="button" hx-get="{% url 'berichts-form' %}" hx-target="#berichtsform" hx-swap="beforeend">`
            neue Berichtsart
        `</button>`
`</div>`
`<div id="berichtsform"></div>`

....
`{% endblock content %}`

#########################
2. Step cancel. This will delete in template bericht.html `<div id="berichtsform"></div>`
Step 1 no longer work.

<-- berichts_form.html -->
...
        `<button hx-get="." hx-target="#berichtsform" hx-swap="delete" type="submit">`
            Cancel
        `</button>`
...

**Comment by rag57** [comment]
`def abbruch(request):`
`    return HttpResponseClientRefresh()`

`<button hx-get="{% url 'abbruch' %}" type="submit">Cancel</button>`

This do the job for the moment.
